### ISSUE TITLE
ProfileFileSupplier.defaultSupplier() leads to NullPointer when config files are missing

### ISSUE THREAD

**Issue by maschnetwork** [aws/aws-sdk-java-v2#5635]
### Describe the bug

We recently introduced automatic credentials reload in the [EventBridge Kafka connector](https://github.com/awslabs/eventbridge-kafka-connector) by leveraging _ProfileFileSupplier.defaultSupplier_  as recommended in the [official docs](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials-profiles.html#profile-reloading):

```java
var provider = DefaultCredentialsProvider.builder()
         //.profileFile(ProfileFile.defaultProfileFile()) <- this is the default by the SDK 
          .profileFile(ProfileFileSupplier.defaultSupplier()) //<- This supports automatic reload
          .build();

var creds = provider.resolveCredentials();
```

However, when no credential files are present (_~/.aws/credentials_ and _~/.aws/config_ are missing) the DefaultCredentialProviderChain.resolveCredentials() process [will result in NullPointer Exceptions](https://github.com/aws/aws-sdk-java-v2/blob/832c1b93a2c2712e65d03f687d11fde8a8a6faa2/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFile.java#L314) in both ProfileCredentialsProvider and InstanceProfileCredentialProvider: `java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null`. (See Additional Information below).

This will happen first in the `ProfileCredentialsProvider` and will be ignored to move to the next provider as outlined [here](https://github.com/aws/aws-sdk-java-v2/blob/3c6c397238870c423bd5e9cb30c1bbce1a3615a7/core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java#L116). But the process will run into it again for the _InstanceProfileCredentialProvider_, for example in [Ec2MetaDataConfigProvider](https://github.com/aws/aws-sdk-java-v2/blob/3c6c397238870c423bd5e9cb30c1bbce1a3615a7/core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/Ec2MetadataConfigProvider.java#L129). 

This behavior essentially influences (and breaks) the DefaultCredentialsProvider chain. It behaves differently than the current SDK default value `ProfileFile.defaultProfileFile()` which does not produce any exceptions and lets _InstanceProfileCredentialProvider_ resolve even though files are missing.

**Related issues:**

https://github.com/awslabs/eventbridge-kafka-connector/issues/396

### Expected Behavior

_ProfileFileSupplier.defaultSupplier()_ behaves the same as  _ProfileFile.defaultProfileFile()_ when credential files are missing.

### Current Behavior

_ProfileFileSupplier.defaultSupplier()_ leads to Nullpointer during resolution process and breaks _DefaultCredentialsProvider_ (or at least behaves different than the default)

### Reproduction Steps

1. First, delete local files such as _~/.aws/credentials_ and _~/.aws/config_
2. Run the below code with the default value of ProfileFile.defaultProfileFile()
3. Run the below code with the non-default value: profileFile(ProfileFileSupplier.defaultSupplier()

```java
public class Main {
    public static void main(String[] args) {
        var provider = DefaultCredentialsProvider.builder()
                    //    .profileFile(ProfileFile.defaultProfileFile()) <- This is the default and is used implicit
                        .profileFile(ProfileFileSupplier.defaultSupplier()) // <- This produces NullPointer when no credentials files are present (Check Errors in the ProviderChain) 
                        .build();

        var creds = provider.resolveCredentials();
        System.out.println(creds);
    }
}
```

4. Check for Nullpointer in the resolution process and see different path for InstanceProfileCredentialsProvider 

### Possible Solution

When no credential files are present (_~/.aws/credentials_ and _~/.aws/config._ are missing) 
 [ProfileFileSupplier.defaultSupplier() effectively returns ](https://github.com/aws/aws-sdk-java-v2/blob/832c1b93a2c2712e65d03f687d11fde8a8a6faa2/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileSupplier.java#L58)`ProfileFile.builder().build()`since _credentialsSupplierOptional_ and _configurationSupplierOptional_ are empty. Later during resolveCredentials of the DefaultCredentialsProviderChain the _contentLocation_ within ProfileFile Builder will be null and therefore [lead to the Nullpointer](https://github.com/aws/aws-sdk-java-v2/blob/832c1b93a2c2712e65d03f687d11fde8a8a6faa2/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFile.java#L314):

```java
@Override
public ProfileFile build() {
    InputStream stream = content != null ? content :
                         FunctionalUtils.invokeSafely(() -> Files.newInputStream(contentLocation));

```

It would make sense to provide a sane default here such as ProfileFile.defaultProfileFile().

### Additional Information/Context

This will happen first in the `ProfileCredentialsProvider` and will be ignored to move to the next provider as outlined [here](https://github.com/aws/aws-sdk-java-v2/blob/3c6c397238870c423bd5e9cb30c1bbce1a3615a7/core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java#L116). But InstanceProfileCredentialProvider will run into it again, for example in [Ec2MetaDataConfigProvider](https://github.com/aws/aws-sdk-java-v2/blob/3c6c397238870c423bd5e9cb30c1bbce1a3615a7/core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/Ec2MetadataConfigProvider.java#L129).

### AWS Java SDK version used

2.28.8

### JDK version used

17

### Operating System and version

MacOS

**Comment by debora-ito** [comment]
Acknowledged. We should fix the NullPointerException.

**Comment by embano1** [comment]
@debora-ito do you have any timelines in mind for when to expect the fix?

**Comment by yomal** [comment]
I also agree with the solution @maschnetwork has mentioned, that we could assign a the `ProfileFile.defaultProfileFile()` for the supplier to resolve. Can you assign this to me so I can start working on a PR?

**Comment by embano1** [comment]
@yomal @debora-ito can you please coordinate so @yomal can start working on this?
